### ISSUE TITLE
Add attention sink to flash attention

### ISSUE THREAD

**Issue by RissyRan** [jax-ml/jax#30964]
Hi team,

Could we add attention sink mechanism to flash attention, to enable gpt-oss model support?

I think we should add something like bellow after this qk_product `logits = jnp.einsum("sd,td->st", q.astype(jnp.float32), k.astype(jnp.float32))` [here](https://github.com/jax-ml/jax/blob/8514d7b4a44b67877166b9b697768bca3b303f8a/jax/experimental/pallas/ops/tpu/splash_attention/splash_attention_kernel.py#L185).

Reference implementation [here](https://github.com/huggingface/transformers/blob/main/src/transformers/models/gpt_oss/modular_gpt_oss.py#L214-L220) , and main logic:

```
sinks = module.sinks.reshape(1, -1, 1, 1).expand(query.shape[0], -1, query.shape[-2], -1)
combined_logits = torch.cat([attn_weights, sinks], dim=-1)
combined_logits = combined_logits - combined_logits.max(dim=-1, keepdim=True).values
probs = F.softmax(combined_logits, dim=-1, dtype=combined_logits.dtype)
scores = probs[..., :-1]  # we drop the sink here
```

Please:

- [x] Check for duplicate requests.
- [x] Describe your goal, and if possible provide a code snippet with a motivating example.

**Comment by justinjfu** [comment]
Closing this issue since the PR has been merged.
